### DISCUSSION TITLE
Singularity container recipe

### DISCUSSION THREAD

**Discussion by JoelRV** [BayraktarLab/cell2location#72] [Q&A] [answered]
Hi there,

Would it be possible to have a reference link to the singularity recipe file used to generate the container you guys linked (https://cell2location.cog.sanger.ac.uk/singularity/cell2location-v0.05-alpha.sif).

I would like to test the build process on my side and eventually add some custom steps, but to be sure that I can reproduce your container, the ".def" file would be a great help.

Thank you in advance for any help.

Best regards!

**Comment by vitkl** [comment]
Hi @JoelRV 

Thanks for using cell2location! 

Singularity container is created from Quay docker container:
```bash
docker pull quay.io/vitkl/cell2location
```
with this command:
```bash
sudo singularity build cell2location.sif docker://quay.io/vitkl/cell2location:latest
```

You can start building from it like this:
```
FROM quay.io/vitkl/cell2location:latest
```

The recipe for Docker is here: https://github.com/BayraktarLab/cell2location/blob/master/Dockerfile

**Comment by JoelRV** [comment]
Hi @vitkl ,

Thanks for getting back to me so fast and your great work with cell2location!

The info and the docker recipe should help me debug my container.

Since there have been changes for pymc3, theano and the newer theano-pymc package, I am trying to get your package running smoothly with the nightly builds of pytorch, since I need it to run it with our newer 3090 nvidia graphic cards.

**Comment by vitkl** [comment]
Hi @JoelRV 

We are finalising a major revision of cell2location including backend (scvi-tools & pyro) and user interface. Please have a look at the new installation instructions and a new tutorial. Switching to scvi-tools addresses the issue we are discussing here. 

Would be also good to get any thoughts or feedback on the new interface and tutorials in discussions: https://github.com/BayraktarLab/cell2location/discussions/65.

**Comment by JoelRV** [comment]
Hey @vitkl ,
Great to see you guys improving the usability! For the moment I am still using the previous release (don't break stuff that works for the moment), but I will try to test the singularity container in my cluster environment and get back to you if things are easier to get running on 3090 gpus without tweaking.

**Comment by vitkl** [comment, marked as answer]
The new version relies on pyro and pytorch as a backend, and can be used
without singularity (in fact the latest singularity image does not include
this version until we release as a pip package).

**Comment by vitkl** [comment]
I will move this to discussions to keep the thread open and since it's not an issue with cell2location.
